src/libsync/owncloudpropagator.cpp

```diff
@@ -1029,6 +1029,7 @@ void CleanupPollsJob::slotPollFinished()
             deleteLater();
             return;
         }
+        _journal->setUploadInfo(job->_item->_file, SyncJournalDb::UploadInfo());
```

Comment:
Can PollJobs have a "there was an error, don't retry" result in any way? I'm just wondering that no other branch wipes the uploadinfo.

PollJob seems to wipe the pollinfo on non-fatal non-503

Reply:
Speaking of `PollInfo`: It's set on `startPollJob` but only removed in this specific error case. And then when a sync is started, PollJobs are rerun? That means polls run indefinitely and block further syncs as long as the server replies "init" or "started"?

I think the idea is that we can continue polling for completion in case the user aborts the sync? (I think you tested that `abort()` case in the unittest, excellent!) PollInfos need more documentation about their purpose.

Reply:
In case of error, the uploadinfo has to stay so the next sync will deal with it by retrying the final move, or use the upload info to resolve possible conflict, should the file have been updated after all (using checksum)

Reply:
Yes, a server never finishing that always reply "started" will block the sync forever.
Do you think this is a problem?

Reply:
UploadInfo not being wiped on failure: So if the server wants to communicate "something failed permanently, stop polling", the poll job would fail, we'd try to resend the final MOVE during the next sync and there it'd fail. Alright, that works.

Infinite polling if server replies that way: Should be ok, would be a significant server bug if that happens and working around it is complicated.